redisson/src/main/java/org/redisson/command/CommandAsyncService.java

```diff
@@ -112,6 +112,7 @@ public void syncSubscriptionInterrupted(RFuture<?> future) throws InterruptedExc
         try {
             future.await();
         } catch (InterruptedException e) {
+            future.cancel(true);
```

Comment:
Ensures the future is cancelled and interrupted in case the caller thread gets interrupted.